cpp/src/fil/common.cuh

```diff
@@ -63,16 +63,19 @@ struct dense_tree {
 /** dense_storage stores the forest as a collection of dense nodes */
 struct dense_storage {
   __host__ __device__ dense_storage(dense_node* nodes, int num_trees,
-                                    int tree_stride, int node_pitch)
+                                    int tree_stride, int node_pitch,
+                                    val_t* vector_leaf)
```

Comment:
why is `vector_leaf` a `val_t*`? Aren't you only using it to store floats? Otherwise, wouldn't the code need a normalizing term per node and some in-kernel math for that?

Reply:
actually, storing ints would be a bad idea, since it wouldn't raise the precision or determinism of the prediction, but would cost an extra lookup, even if it hits L1.

Reply:
I was just mirroring what was used in the existing tree structure. Can definitely change it to floats if there is no other use case.